src/dbseeder/programs.py

```diff
@@ -134,116 +144,192 @@ class WqpProgram(object):
 
-        results_folder = join(parent_folder, 'Results')
-        stations_folder = join(parent_folder, 'Stations')
+            if not isdir(parent_folder):
+                raise Exception('Pass in a location to the parent folder that contains WQP. {}'.format(parent_folder))
 
-        if not isdir(results_folder or stations_folder):
-            raise Exception('WQP folder is missing Results or Stations child folders containing csv files.')
+            results_folder = join(parent_folder, 'Results')
+            stations_folder = join(parent_folder, 'Stations')
 
-        #: all is well set the folders for seeding later on
-        self.results_folder = results_folder
-        self.stations_folder = stations_folder
+            if not isdir(results_folder or stations_folder):
+                raise Exception('WQP folder is missing Results or Stations child folders containing csv files.')
 
-        self.db = db
+            #: all is well set the folders for seeding later on
+            self.results_folder = results_folder
+            self.stations_folder = stations_folder
 
     def seed(self):
+        if not hasattr(self, 'results_folder') or not hasattr(self, 'stations_folder'):
+            raise Exception('You must pass a file location if you are seeding. Did you want to update?')
+
+        try:
+            self._seed_by_file()
+        finally:
+            if hasattr(self, 'cursor'):
+                del self.cursor
+
+    def update(self):
+        try:
+            last_updated = self._get_most_recent_result_date()
+
+            if not last_updated:
+                raise Exception('No last updated date')
+
+            #: get new results from wqp service
+            result_url = self._format_url(self.wqp_url, 'Result', last_updated)
+            #: group them as if they were read from querycsv
+            new_results = self._group_rows_by_id(HttpClient.get_csv(result_url))
+
+            #: find the station ids from the new results that aren't in the database
+            new_station_ids = self._find_new_station_ids(new_results)
+
+            if new_station_ids and len(new_station_ids) > 0:
+                print('of the new stations found, inserting {} unique'.format(len(new_station_ids)))
+
+                station_url = self._format_url(self.wqp_url, 'Station', last_updated)
+                new_stations = HttpClient.get_csv(station_url)
+
+                if not new_stations:
+                    return
+
+                header = new_stations.next()
+
+                stations = self._extract_stations_by_id(new_stations, new_station_ids, header)
+
+                #: TODO do we need to do anything with wxps here?
+                self._seed_stations(stations, header=header, wxps=[])
+            else:
+                print('all stations already in database')
+            for samples_for_id in new_results.values():
+                self._seed_results(samples_for_id)
```

Comment:
I think that you should do a similar check for existing data (using the `SampleId` field?) for results that you do with stations above to prevent duplicate data from polluting the database. I would expect that an update process should be able to be run more than once per day without negative effects.